service/history/transferQueueActiveTaskExecutor.go

```diff
@@ -324,17 +326,16 @@ func (t *transferQueueActiveTaskExecutor) processCloseExecution(
 			},
 			CompletionEvent: completionEvent,
 		})
-
-		// Check to see if the error is non-transient, in which case reset the error and continue with processing
-		if _, ok := err.(*serviceerror.NotFound); ok {
-			err = nil
+		switch err.(type) {
+		case nil:
```

Comment:
This checks for type being `nil` not value. What's wrong with former version with `if`?

Reply:
in general, if there is special treatment to error based on type, doing a switch case will make the handling logic much more clear